pwndbg/commands/heap.py

```diff
@@ -182,6 +182,41 @@ def heap(addr=None, verbose=False, simple=False) -> None:
         for chunk in h:
             malloc_chunk(chunk.address, verbose=verbose, simple=simple)
 
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.RawTextHelpFormatter,
+    description="""Try to find if an address is belond to a heap chunk.
+
+If yes -- print chunk. Search all heaps.""",
+)
+parser.add_argument(
+    "addr",
+    type=int,
+    help="Address of the interest.",
+)
+parser.add_argument(
+    "-v", "--verbose", action="store_true", help="Print all chunk fields, even unused ones."
+)
+parser.add_argument(
+    "-s", "--simple", action="store_true", help="Simply print malloc_chunk struct's contents."
+)
+
+
+@pwndbg.commands.ArgparsedCommand(parser, category=CommandCategory.HEAP)
+@pwndbg.commands.OnlyWhenRunning
+@pwndbg.commands.OnlyWithResolvedHeapSyms
+@pwndbg.commands.OnlyWhenHeapIsInitialized
+def hi(addr, verbose=False, simple=False) -> None:
+    """Iteratively search all heaps for contain current address in their chunks bodys.
+    """
+    allocator = pwndbg.heap.current
+    arenas = allocator.arenas
+
+    for arena in arenas:
+        for heap in arena.heaps:
+            if heap.start <= addr and heap.end > addr:
+                for chunk in heap:
+                    if chunk.address <= addr and (chunk.address + chunk.size) > addr:
+                        malloc_chunk(chunk.address, verbose=verbose, simple=simple)
```

Comment:
Is it possible that the address is in more than one chunk?
If not, maybe we could return after the first match to stop iterating further?

Reply:
Sorry for delay.

It totally relay on how pwndbg heap impl work.
And, as you notice, in likely() cases it will print only first one.
If after mchunk_size corruption pwndbg still able to display several chunks -- it will print them too.

I think we can add option ```continue=False``` by default.